core/domain/suggestion_services.py

```diff
@@ -322,40 +321,59 @@ def accept_suggestion(suggestion, reviewer_id, commit_message, review_message):
                 suggestion.suggestion_id)
         )
 
+    suggestion.set_suggestion_status_to_accepted()
+    suggestion.set_final_reviewer_id(reviewer_id)
+
     author_name = user_services.get_username(suggestion.author_id)
     commit_message = get_commit_message_for_suggestion(
         author_name, commit_message)
-    mark_review_completed(
-        suggestion, suggestion_models.STATUS_ACCEPTED, reviewer_id)
     suggestion.accept(commit_message)
-    thread_id = suggestion.suggestion_id
+
+    _update_suggestion(suggestion)
+
     feedback_services.create_message(
-        thread_id, reviewer_id, feedback_models.STATUS_CHOICES_FIXED,
+        suggestion_id, reviewer_id, feedback_models.STATUS_CHOICES_FIXED,
         None, review_message)
 
+    # When recording of scores is enabled, the author of the suggestion gets an
+    # increase in their score for the suggestion category.
     if feconf.ENABLE_RECORDING_OF_SCORES:
-        increment_score_for_user(
-            suggestion.author_id, suggestion.score_category,
-            suggestion_models.INCREMENT_SCORE_OF_AUTHOR_BY)
+        user_id = suggestion.author_id
+        score_category = suggestion.score_category
+
+        # Get user scoring domain object.
+        user_scoring = get_user_scoring(user_id, score_category)
+        if user_scoring is None:
+            # Create the user scoring model and domain object.
+            user_scoring = create_new_user_scoring(user_id, score_category, 0)
+
+        # Increment the score of the author due to their suggestion being
+        # accepted.
+        user_scoring.increment_score(
+            suggestion_models.INCREMENT_SCORE_OF_AUTHOR_BY
+        )
+
+        # Emails are sent to onboard new reviewers. These new reviewers are
+        # created when the score of the user passes the minimum score required
+        # to review.
         if feconf.SEND_SUGGESTION_REVIEW_RELATED_EMAILS:
-            scores = get_all_scores_of_user(suggestion.author_id)
-            if (
-                    suggestion.score_category in scores and
-                    scores[suggestion.score_category] >=
-                    feconf.MINIMUM_SCORE_REQUIRED_TO_REVIEW):
-                if not check_if_email_has_been_sent_to_user(
-                        suggestion.author_id, suggestion.score_category):
-                    email_manager.send_mail_to_onboard_new_reviewers(
-                        suggestion.author_id, suggestion.score_category)
-                    mark_email_has_been_sent_to_user(
-                        suggestion.author_id, suggestion.score_category)
-
-
-def reject_suggestion(suggestion, reviewer_id, review_message):
-    """Rejects the suggestion.
+            if user_scoring.user_can_review_the_category() and (
```

Comment:
Method names should start with verbs. Please ensure that that is the case throughout.

Reply:
Why not just can_user_review_the_category()?